Prepublishing publish button cut off on some devices

jd-alexander opened the pull request:
Fixes #12491 

## Findings
Android 10, introduced enhanced System Navigation. That basically allows you to select if you want to utilize navigation buttons or gestures to navigate your phone. 

<kbd><img src="https://user-images.githubusercontent.com/1509205/116342402-2e6a6100-a7a8-11eb-9e6e-2ed3aa226920.png" width="320"></kbd>

This functionality broke how the `PrepublishingBottomSheet` was being rendered in the `StoryComposerActivity` because the `StoryComposerActivity` has `IMMERSIVE_MODE` and `FULLSCREEN` styles enabled so that the Story Composer can take up the entire screen. This has led us to create custom mechanisms for determining the bottom margin of the bottom sheet's root container. 

## Solution

The solution is to determine the appropriate bottom margin depending on the navigation mode of the device. In the button mode, the existing bottom inset works well with the view's bottom margin. However, in gesture mode, the bottom inset is basically at the bottom edge of the screen, so we simulate a navigation status bar height and add that to the bottom margin so that the bottom sheet will look similar to how it looks in 3 button navigation mode. 

**Concerns**

* To enable the bottom margin modification of the bottom sheet I decided to utilize the `isStoryPost` flag that's passed to the `Fragment` to determine if immersive mode is on or not. 
https://github.com/wordpress-mobile/WordPress-Android/blob/08045f3df77c84423ab1cd514bfa4a7f5641fe72/WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingBottomSheetFragment.kt#L141
Currently, the `StoryComposerActivity` has this configuration as a style 
https://github.com/wordpress-mobile/WordPress-Android/blob/f109fd172bea27a5554a69fed54e8581c622baab/WordPress/src/main/res/values/styles.xml#L217
and when I go within the `Fragment` to access this via the `decorView.systemUiVisibility == View.SYSTEM_UI_FLAG_IMMERSIVE` this returned false, so I am thinking this may be the case because these values were not set programmatically. Not sure.

* In gesture mode, the status bar height that's utilized does not make the padding below and above the Publish button equal. I haven't figured out how to get that better in a standardized way because I did not want to utilize an arbitrary value. Nonetheless, we could lessen the status bar height since this solution is the best approach we have so far.

* The `WindowInsetListener` was behaving a bit unpredictable so I opted not to use it. Also, it was triggering the bottom sheet's inset when the keyboard is shown on the screen. I remember seeing an issue where the keyboard would still overlay the bottom sheet occasionally, but I am not sure, so I could double-check that after the first test and review of this PR is done.

## Testing

**Pre-requisite** 

A device or emulator running Android 10 and above with gesture navigation enabled. To Enable Gesture Mode go to Settings -> Accessibility -> System Navigation -> Select Gesture navigation.

**Steps**
1. Create a new Story Post. 
2. Click Publish. 
3. Notice that the bottom sheet is no longer cut off. 

Before | After 
--------|-------
<kbd><img src="https://user-images.githubusercontent.com/1509205/116344423-d0d81380-a7ab-11eb-9d39-f84ba32b71f9.png" width="320"></kbd>       |   <kbd><img src="https://user-images.githubusercontent.com/1509205/116345672-33321380-a7ae-11eb-92f1-2de8be339748.png" width="320"></kbd>

## Regression Notes

### 1. Potential unintended areas of impact

The `PrepublishingBottomsheet` component is utilized in several activities, so all three usages were checked to ensure that the bottom sheet is properly positioned within the view. 

### 2. What I did to test those areas of impact (or what existing automated tests I relied on)

a. PostsListActivity 

1. Open Post List. 
2. Go to the Drafts tab. 
3. Click Publish on a draft. 
4. Notice the bottom sheet component had the appropriate bottom margin. 

b. EditPostActivity

1. Open the Post Editor. 
2. Click the Publish button. 
3. Notice the bottom sheet component has the appropriate bottom margin. 

PostsListActivity | EditPostActivity
--------|-------
  <kbd><img src="https://user-images.githubusercontent.com/1509205/116342769-c8320e00-a7a8-11eb-9b2e-9d00ac095194.png" width="320"></kbd>  |  <kbd><img src="https://user-images.githubusercontent.com/1509205/116342772-c9fbd180-a7a8-11eb-98d7-fe79b0cab0f4.png" width="320"></kbd>
     
### 3. What automated tests I added (or what prevented me from doing so)

I didn't add any automation tests since the behavior modification was UI-centric and not necessarily user flow-related. 


PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

jd-alexander committed:
aded a nav bar mode helper to determine if device is in gesture mode.

jd-alexander committed:
integrated gesture mode in the bottom sheet component.

jd-alexander committed:
Added a comment about the solution.

jd-alexander committed:
Updated comments and only enable this logic on story posts.

jd-alexander committed:
Fixed lint error.

jd-alexander committed:
reduced nesting within the addWindowInsetToFragmentContainer function.

jd-alexander committed:
added link to Google code that inspired the nav bar helper.

jd-alexander committed:
Since marshmallow the status bar height is 24dp.

jd-alexander commented on WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingBottomSheetFragment.kt:
```diff
@@ -119,15 +125,37 @@ class PrepublishingBottomSheetFragment : WPBottomSheetDialogFragment(),
+                                0,
+                                0,
+                                0,
+                                windowInsets.systemGestureInsets.bottom + dpToPx(
```
I utilized the `systemGestureInsets` because it was returning an inset that represents the touchable area for gestures. Using `systemWindowInsetBottom` here had resulted in an inset that was really low, which I think is due to the fact that the host/parent activity is in immersive mode.

aforcier commented:
Thanks @jd-alexander ! I gave this a try, here's what I found:

1. Pixel 5 - Gesture navigation on: ✅ 
2. Pixel 5 - 3-button navigation on: ✅ 
3. Pixel 4 - Gesture navigation on: ❓ 
4. Pixel 3 - 3-button navigation: ❓ 

It looks like on the Pixel 4 we're overcounting the status bar height, so the bottom padding for the prepublish sheet is a bit too much:

Keyboard shown:
![pixel-4-keyboard-up](https://user-images.githubusercontent.com/9613966/116355606-08d76900-a835-11eb-81bf-017fd45e9e0a.png)

Keyboard hidden:
![pixel-4-bottom-sheet-spacing](https://user-images.githubusercontent.com/9613966/116355354-b0a06700-a834-11eb-8ebc-6f344bd852d4.png)

The same seems true for the Pixel 3, though there it's not as noticeable because with the keyboard present the bottom sheet is too tall and gets cropped at the bottom anyway:

Keyboard shown:
![pixel-3-keyboard-up](https://user-images.githubusercontent.com/9613966/116355784-4fc55e80-a835-11eb-8650-677d234d51da.png)

Keyboard hidden:
![pixel-3-keyboard-down](https://user-images.githubusercontent.com/9613966/116355798-5653d600-a835-11eb-94f5-405acb787d39.png)

jd-alexander commented:
Thanks @aforcier I am checking this out to see how this PR can be improved to resolve the issue.

jd-alexander committed:
Merge branch 'develop' into issue-12491/prepublishing-cut-off-on-some-devices

jd-alexander committed:
Increased margin bottom of fragment container on large screens.

jd-alexander committed:
Fixed lint errors.

jd-alexander committed:
remove submodule change

aforcier commented:
@jd-alexander I gave the latest changes a try. The results were better, though there's still an issue on the Pixel 4:

1. Pixel 5 - Gesture navigation on: ✅
2. Pixel 5 - 3-button navigation on: ✅
3. Pixel 4 - Gesture navigation on: ❌ 
4. Pixel 4 - 3-button navigation on: ❌ 
5. Pixel 3 - 3-button navigation: ✅

The Pixel 4 still has the extra padding as before. The dimensions of the PIxel 4 are 1080 x 2280 - I thought maybe the `MIN_SCREEN_HEIGHT_IN_PX` needed a slightly larger value, so I tried `2280` and that seems to be working on all devices. I'm not sure if that might be having some other repercussions I'm not thinking of though. Happy to try out something else if you'd like!

I wanted to mention I also feel like I'm seeing more of the occasional issue I believe you've spent some time on before, where the keyboard is shown after the bottom sheet and hides most of it. Not sure if that might be related to the changes or not.

jd-alexander commented:
> @jd-alexander I gave the latest changes a try. The results were better, though there's still an issue on the Pixel 4:

Thanks for testing and reviewing so far @aforcier I appreciate it!

> The Pixel 4 still has the extra padding as before. The dimensions of the PIxel 4 are 1080 x 2280 - I thought maybe the MIN_SCREEN_HEIGHT_IN_PX needed a slightly larger value, so I tried 2280 and that seems to be working on all devices. 

Thanks! It was working fine on the Pixel 4 I was testing but I think for some reason the screen height I was getting was less than the amount you mentioned hence the incorrect MIN_SCREEN_HEIGHT_IN_PX. 

> I'm not sure if that might be having some other repercussions I'm not thinking of though. Happy to try out something else if you'd like!

This is the best solution I can think of right now. I spent some time looking into how we categorize DPIs because that felt better than hardcoding screen height itself but I didn't have a way to determine the DPI of the Pixel 5, I ended up creating an emulator with the same screen resolution and the DPI values I saw were of `xxhdpi `so I wasn't able to use the DPI value. I am wondering if this is something you could check for me. 

<img width="660" alt="Screenshot 2021-04-30 at 12 26 24 PM" src="https://user-images.githubusercontent.com/1509205/116731568-77d8cd00-a9af-11eb-9133-2e6bcdff61be.png">

> I wanted to mention I also feel like I'm seeing more of the occasional issue I believe you've spent some time on before, where the keyboard is shown after the bottom sheet and hides most of it. Not sure if that might be related to the changes or not.

I was able to reproduce this. Thanks for mentioning it. I am gonna have to check to see if it's related. It probably isn't and the logic needs to be improved somehow. 

@mzorz I wanted to loop you in on this PR whenever you get a chance, just to take a look at this solution, as we are trying to optimize the bottom sheet's behavior once and for all 😄

mzorz commented:
> @mzorz I wanted to loop you in on this PR whenever you get a chance, just to take a look at this solution, as we are trying to optimize the bottom sheet's behavior once and for all 😄

Thank you for the ping @jd-alexander ! 👍 - I think overall deciding on which version we're running in order to use one minimum offset or the other is a good enough approach, the problem has been chasing us and it's difficult to know how to calculate it best. I do wonder what happens if we use different keyboard (3rd party software instead of the stock keyboard) which can potentially have different heights, but it'd lead to the same problem. It's probably enough to go with the stock keyboard as test and take it from there, leaving it in a better state.

I've tested on a ![Screenshot_20210503-182959](https://user-images.githubusercontent.com/6597771/116904788-87a51b00-ac14-11eb-9606-6803b83a5e72.jpeg) of handsets here:
- Pixel 4a: ✅ (both with and without the navigation buttons)
- Pixel 2 (1080x1920): ❓ this one is showing a cut, probably  not related to this PR, it occupies more than what the screen can show.
Here's 2 screenshots with the keyboard shown and hidden:

![Screenshot_20210503-180607](https://user-images.githubusercontent.com/6597771/116903704-216bc880-ac13-11eb-8540-5ed1ed75fc52.png)
![Screenshot_20210503-180616](https://user-images.githubusercontent.com/6597771/116903706-22045f00-ac13-11eb-9b56-6865485ac273.png)
- Samsung J2 (540x960): ❓ this is unfortunately the worst case 😕 , it is barely usable, see the video I recorded. The text is too far up and I can't see what I type, and the `Publish` button is totally hidden behind the keyboard.

https://user-images.githubusercontent.com/6597771/116904459-19605880-ac14-11eb-9f89-f32c9f6c9c9e.mp4

A few screenshots:
![Screenshot_20210503-183015](https://user-images.githubusercontent.com/6597771/116904784-8673ee00-ac14-11eb-85e9-ea3366b37272.jpeg)
![Screenshot_20210503-182948](https://user-images.githubusercontent.com/6597771/116904786-870c8480-ac14-11eb-829a-52b3830d7ab8.jpeg)


I know this was not exactly what we were looking to fix but it looks like we may need a different version of the publishing sheet for phones with smaller screens maybe?

jd-alexander commented:
> Thank you for the ping @jd-alexander ! 👍 - I think overall deciding on which version we're running in order to use one minimum offset or the other is a good enough approach, the problem has been chasing us and it's difficult to know how to calculate it best. I do wonder what happens if we use different keyboard (3rd party software instead of the stock keyboard) which can potentially have different heights, but it'd lead to the same problem. It's probably enough to go with the stock keyboard as test and take it from there, leaving it in a better state.

Thanks for such a thorough breakdown above @mzorz . I really appreciate it. Reminding me about the 3rd party keyboards is so important because sometimes when testing, I forget that there's quite a lot of variation out there in terms of keyboard usage. I will definitely keep that in mind when I dive further into the issue. 

Due to some of the layout issues taking place, I may need to create `dimens` values that target the different DPIs, so I will be spinning up several emulators to reproduce the issues described here as best as possible. For issues, I am not able to reproduce we may need to do a pair programming session to dig into what is taking place on the physical devices.

mzorz commented:
Thank you for your comment @jd-alexander !
> Due to some of the layout issues taking place, I may need to create dimens values that target the different DPIs, so I will be spinning up several emulators to reproduce the issues described here as best as possible. For issues, I am not able to reproduce we may need to do a pair programming session to dig into what is taking place on the physical devices.

For sure, let me know and we can schedule something as needed. I can try reproducing here on other emulators as well as needed 👍

antonis commented:
Hey @jd-alexander 👋 
great work tackling this. Your approach seems reasonable and works in most cases.
I tested this on a Pixel 2 XL and a Pixel 5 both with Android 11.
The app works as expected for stories in my tests which is a big improve on Pixel 5.
I only noticed an issue the first time I press publish in simple posts on the same device.

https://user-images.githubusercontent.com/304044/117468004-ee9c3a00-af5c-11eb-90de-17e89bff46ca.mp4

I'll be happy to help with testing this or otherwise.

My impression from working on the [Site Design and Page Layout pickers](https://github.com/wordpress-mobile/WordPress-Android/blob/70ce7de8817020d29768074bb408688436320220/WordPress/src/main/java/org/wordpress/android/ui/FullscreenBottomSheetDialogFragment.kt) is that the `BottomSheetDialogFragment` implementation is a bit buggy 😐

AliSoftware commented:
Moving this PR to next milestone, since 17.6 goes into code freeze today.

_This seems to have been stale for >1 month, could be worth pinging folks to help finally get this merged for the next one?_

ashiagr committed:
Merge branch 'develop' into issue-12491/prepublishing-cut-off-on-some-devices

ashiagr commented:
👋 @antonis 

Getting back on this issue, I merged latest `develop` and tested it on the `Pixel 5`, `Android 11` but wasn't able to reproduce the below scenario: 

> I only noticed an issue the first time I press publish in simple posts on the same device.

Here's how I see it

https://user-images.githubusercontent.com/1405144/127963792-e6f70aab-0a56-4222-a898-7aa3b5379eec.mp4

Can you retry and confirm you still see this issue?

antonis commented:
Hey @ashiagr 👋 

> Can you retry and confirm you still see this issue?

I tested the latest update from this branch after the merge and I'm not able to reproduce the [reported issue](https://github.com/wordpress-mobile/WordPress-Android/pull/14533#issuecomment-834485322) either 🎉

ashiagr commented:
👋 @aforcier 

Based on your https://github.com/wordpress-mobile/WordPress-Android/pull/14533#issuecomment-829856234, 

> The Pixel 4 still has the extra padding as before. The dimensions of the PIxel 4 are 1080 x 2280 - I thought maybe the MIN_SCREEN_HEIGHT_IN_PX needed a slightly larger value, so I tried 2280 and that seems to be working on all devices.

I tested on `Pixel4`, `Android 11` with existing `MIN_SCREEN_HEIGHT_IN_PX = 2200` after merging the latest `develop`. 

<img width="500" alt="Screenshot 2021-08-04 at 10 29 23 AM" src="https://user-images.githubusercontent.com/1405144/128125808-b58e6ceb-8b1c-4e91-b101-66004eaa89bd.png">

Here's how I see it:

Pixel4 3-Button-Nav (2200) | Pixel4 Gesture-Nav (2200)
-----------------|---------------------
![pixel_4_3_btn](https://user-images.githubusercontent.com/1405144/128125415-79e6fa3e-b35b-4283-a6d6-73261d274491.png) | ![pixel_4_gesture_nav](https://user-images.githubusercontent.com/1405144/128125430-5818cc29-426a-4ce2-a803-b3f1fa844145.png)

With `MIN_SCREEN_HEIGHT_IN_PX = 2280` I see the publish button is not visible when the keyboard is visible:

Pixel4 3-Button-Nav (2280) | Pixel4 Gesture-Nav (2280)
-----------------|---------------------
![pixel_4_3_btn_2280](https://user-images.githubusercontent.com/1405144/128125570-25fbf1d4-404e-4720-ae7b-8f9e2dbb9f7f.png) | ![pixel_4_gesture_nav_2280](https://user-images.githubusercontent.com/1405144/128125578-51e8e27d-eb4e-499b-bea2-acf9297813be.png)


Can you give it another try and share your findings?

aforcier commented:
Hey @ashiagr! I gave this a try but unfortunately the behavior is unchanged from my [last test](https://github.com/wordpress-mobile/WordPress-Android/pull/14533#issuecomment-829856234) when testing this branch on a physical Pixel 4. (IIRC Joel and I noticed some inconsistencies between emulators and the real thing on some of these tests)

![pixel4-3button](https://user-images.githubusercontent.com/9613966/128283902-e6630775-c2ca-4150-bf9b-0f481f89f63a.png)

![pixel4-gesture](https://user-images.githubusercontent.com/9613966/128283908-c6897586-826f-46d8-ad26-c647bada148e.png)

Continues to work fine on a Pixel 5 (didn't test a Pixel 3 this time).

Also, I messed around with the keyboard settings a bit to match what I'm seeing in your screenshot, e.g. shorter keyboard height and turning on key borders in the keyboard theme, but it didn't change anything.

ashiagr committed:
Reduce pre publishing list item height to improve upon behaviour when keyboard occupies more than what the screen can show

- Reduced top and bottom margin for prepublishing list item with button from 16dp to 12dp - Button already has a top/bottom padding of 4dp, it is sufficient to add 12dp margin to get a 16dp total margin.

- Reduced prepublishing story title thumbnail width and height keeping same aspect ratio of 1.5.

- Reduced prepublishing story title list item 8dp top and bottom padding - Story thumbnail layout already has a 16dp top and bottom padding, reduced extra 8dp + 8dp vertical padding.

- Reduced prepublishing action list item height - Removed font padding on both primary and secondary title and included 2dp top margin only on the secondary title (action_result).

ashiagr committed:
Update diff calculation in keyboard resize view util

Diff is used for adding padding to the pre-publishing bottom sheet content view when the keyboard is visible so that the publish visible is not hidden behind the keyboard. It is approximately equal to the keyboard height.

Based on the runtime values, it seems windowVisibleDisplayFrame height includes the top inset when the screen is in full screen immersive mode and subtracting it from the display height will give a shorter keyboard height. Hence it is subtracted from the real screen size, and also the bottom inset is subtracted to get the actual keyboard height.

ashiagr committed:
Remove add window inset to fragment container logic after revised keyboard height calculation in the previous commit

Checking results on different emulators (gesture/ 3-button-navigation):

Pixel 2: 1080 x 1920
Pixel 3 XL: 1440 x 2960
Pixel 5: 1080 X 2340

this logic doesn't seem to be necessary now.

ashiagr commented:
👋 @aforcier 

Thank you for trying it out and for your patience 🙏.

I investigated it a bit and based on the runtime height values on Pixel 5 on fullscreen immersive mode screen, I revised the keyboard height (diff) calculation slightly in ca83b70. 

<img src="https://user-images.githubusercontent.com/1405144/129517943-5fa6276e-b490-465d-aa4d-ee11e3c86a78.jpg" width = "400">

Also reduced pre-publishing list item height slightly to improve upon behaviour when keyboard occupies more than what the screen can show in e98ba54. E.g. Pixel 2 in https://github.com/wordpress-mobile/WordPress-Android/pull/14533#issuecomment-831383669

I see improved results on different devices/ emulators after these changes:

##### Pixel 5 (API level 29) device (1080 X 2340)

![pixel5out](https://user-images.githubusercontent.com/1405144/129514741-dc1def67-a199-4e82-b5ea-de44a93a7861.png)

##### Pixel 4 (API level 30) emulator (1080 x 2280)

![pixel4out](https://user-images.githubusercontent.com/1405144/129516374-422f0c52-6864-4319-892e-7b5ce199560c.png)

##### Pixel 2 (API level 30) emulator (1080 x 1920)

![pixel2out](https://user-images.githubusercontent.com/1405144/129514731-01c1c4c5-9504-4a5e-a43c-1fa71b380103.png)

##### Pixel 3a (API level 30) device (1080 x 2220)

![pixel3aout](https://user-images.githubusercontent.com/1405144/129515682-5b2a3f85-4d59-4b7d-be72-938403b51fb2.png)

#####  Pixel 3XL (API level 27) emulator (without gesture nav capability) (1440 x 2960)

<img src="https://user-images.githubusercontent.com/1405144/129514915-cde00082-bf5d-4b53-8c3b-a817f2de295f.png" width = 400>

Considering

> IIRC Joel and I noticed some inconsistencies between emulators and the real thing on some of these tests

Can you try out on the devices you have?

------

##### Note for future:

[androidx-core-1.5.0-alpha02 ](https://developer.android.com/jetpack/androidx/releases/core#1.5.0-alpha02) allows checking the visibility and size of the on-screen keyboard. We're currently using [androidx-core-1.3.2](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/build.gradle#L139). Whenever we upgrade this lib, we can utilize the new `WindowInsets` APIs for checking the keyboard (IME) size.

Helpful links: 
- Chris Bane's article on [New WindowInsets APIs for checking the keyboard (IME) visibility and size](https://medium.com/androiddevelopers/animating-your-keyboard-fb776a8fb66d) (under "The IME type" heading)
- [SO answer](https://stackoverflow.com/a/63595830/193545)

aforcier commented:
Nice @ashiagr! I didn't get to this yet but I will try it out tomorrow 👍

aforcier commented:
Okay! I gave this a try and it's working great everywhere, nicely done @ashiagr ! 🎉 

1. Pixel 5 - Gesture navigation: ✅
2. Pixel 5 - 3-button navigation: ✅
3. Pixel 4 - Gesture navigation: ✅ 
4. Pixel 4 - 3-button navigation: ✅ 
5. Pixel 3 - 3-button navigation: ✅

(Testing keyboard up and keyboard down everywhere.)

I didn't dive into the code but it looks like the PR can come out of draft status 😀

ashiagr commented:
Thanks for testing @aforcier! I've marked this PR ready for review. Hopefully, we'll be able to merge it with release `18.1`or the next one🤞.

aforcier commented on WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingBottomSheetFragment.kt:
```diff
@@ -37,6 +36,7 @@ class PrepublishingBottomSheetFragment : WPBottomSheetDialogFragment(),
         PrepublishingScreenClosedListener, PrepublishingActionClickedListener {
     @Inject internal lateinit var viewModelFactory: ViewModelProvider.Factory
     @Inject internal lateinit var analyticsTrackerWrapper: AnalyticsTrackerWrapper
+    @Inject internal lateinit var navBarModeHelper: NavigationBarInteractionModeHelper
```
It looks like `navBarModeHelper` is now unused. In fact, it looks like we can drop the `NavigationBarInteractionModeHelper` class entirely now.

ashiagr replied:
This is now removed in 5fef752

aforcier reviewed (commented):
Went through the changes, other than removing the now-unused `NavigationBarInteractionModeHelper` things seem good to me! I also tested that the prepublishing sheet behavior in the regular post editor is still working correctly.

Might be good to get a quick second look from a more up-to-date WPAndroid developer too, though 😀.

Thanks for adopting this PR @ashiagr 🙂

ashiagr committed:
Remove now unused NavigationBarInteractionModeHelper

ashiagr committed:
Apply AS suggestion

ashiagr committed:
Merge branch 'develop' into issue-12491/prepublishing-cut-off-on-some-devices

antonis reviewed (approved):
Great work @ashiagr and @jd-alexander 👍 
The code changes look consistent too me and the app works as expected 🎉 
I performed tests with different configurations on a Pixel 5 (Android 12) and Pixel 2 XL (Android 11) and was not able to reproduce the issue. Given that we already have [successful tests on other devices](https://github.com/wordpress-mobile/WordPress-Android/pull/14533#issuecomment-901550947) I think this is ready to 🚢
